geth/common/types_test.go

```diff
@@ -0,0 +1,28 @@
+package common
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSendTxArgsValidity(t *testing.T) {
+	// 1. If only data fields is set, valid and return data
+
+	bytes1 := hexutil.Bytes([]byte{0xAA, 0xBB, 0xCC, 0xDD})
+	bytes2 := hexutil.Bytes([]byte{0x00, 0x01, 0x02})
+
+	doSendTxValidityTest(t, SendTxArgs{}, true, nil)
+	doSendTxValidityTest(t, SendTxArgs{Input: bytes1}, true, bytes1)
+	doSendTxValidityTest(t, SendTxArgs{Data: bytes1}, true, bytes1)
+	doSendTxValidityTest(t, SendTxArgs{Input: bytes1, Data: bytes1}, true, bytes1)
+	doSendTxValidityTest(t, SendTxArgs{Input: bytes1, Data: bytes2}, false, nil)
+}
+
+func doSendTxValidityTest(t *testing.T, args SendTxArgs, expectValid bool, expectValue hexutil.Bytes) {
+	assert.Equal(t, expectValid, args.Valid(), "Valid() returned unexpected value")
+	if expectValid {
+		assert.Equal(t, expectValue, args.GetInput(), "Input() returned unexpected value")
```

Comment:
`GetInput()` instead of `Input()`?

Reply:
Oh actually, we could name the getter as `Input()`. This case is described in https://golang.org/doc/effective_go.html#Getters

Reply:
Except it is a syntax error because we have a field named like that 😅